File: ebayfff_ui/src/services/WishListItemService.js
import http from "../http-common";

class WishListItemService {
  getAllWishListItems(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/wishListItem/wishListItems`, searchDTO);
  }

  get(wishListItemId) {
    return this.getRequest(`/wishListItem/${wishListItemId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/wishListItem?field=${matchData}`, null);
  }

  addWishListItem(data) {
    return http.post("/wishListItem/addWishListItem", data);
  }

  update(data) {
  	return http.post("/wishListItem/updateWishListItem", data);
  }
  
  uploadImage(data,wishListItemId) {
  	return http.postForm("/wishListItem/uploadImage/"+wishListItemId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new WishListItemService();

File: src/main/java/com/ebayfff/service/WishListItemService.java
package com.ebayfff.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.ebayfff.domain.WishListItem;
import com.ebayfff.dto.WishListItemDTO;
import com.ebayfff.dto.WishListItemSearchDTO;
import com.ebayfff.dto.WishListItemPageDTO;
import com.ebayfff.dto.WishListItemConvertCriteriaDTO;
import com.ebayfff.service.GenericService;
import com.ebayfff.dto.common.RequestDTO;
import com.ebayfff.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface WishListItemService extends GenericService<WishListItem, Integer> {

	List<WishListItem> findAll();

	ResultDTO addWishListItem(WishListItemDTO wishListItemDTO, RequestDTO requestDTO);

	ResultDTO updateWishListItem(WishListItemDTO wishListItemDTO, RequestDTO requestDTO);

    Page<WishListItem> getAllWishListItems(Pageable pageable);

    Page<WishListItem> getAllWishListItems(Specification<WishListItem> spec, Pageable pageable);

	ResponseEntity<WishListItemPageDTO> getWishListItems(WishListItemSearchDTO wishListItemSearchDTO);
	
	List<WishListItemDTO> convertWishListItemsToWishListItemDTOs(List<WishListItem> wishListItems, WishListItemConvertCriteriaDTO convertCriteria);

	WishListItemDTO getWishListItemDTOById(Integer wishListItemId);







}





